packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js

```diff
@@ -261,13 +261,68 @@ describe('PerformanceMonitoring', function() {
     }, 10)
   })
 
+  it('should adjust transaction start based on earliest span start', function(done) {
+    const firstSpan = new Span('first-span-name', 'first-span')
+    firstSpan.end()
+
+    const transaction = new Transaction('/', 'transaction', {})
+    transaction.onEnd = function() {
+      performanceMonitoring.adjustTransactionTime(transaction)
+      expect(transaction._start).toBe(firstSpan._start)
+      expect(transaction._end).toBeGreaterThanOrEqual(lastSpan._end)
+      done()
+    }
+    transaction.spans.push(firstSpan)
+
+    const lastSpan = transaction.startSpan('last-span-name', 'last-span')
+    lastSpan.end()
+
+    transaction.detectFinish()
+  })
+
+  it('should adjust transaction end based on latest span end', function(done) {
+    const transaction = new Transaction('/', 'transaction', {})
+    const transactionStart = transaction._start
+
+    const firstSpan = transaction.startSpan('first-span-name', 'first-span')
+    firstSpan.end()
+
+    const longSpan = transaction.startSpan('long-span-name', 'long-span')
+
+    const lastSpan = transaction.startSpan('last-span-name', 'last-span')
+    lastSpan.end()
+
+    longSpan.end()
+    longSpan.end += 500
+
+    transaction.onEnd = function() {
+      performanceMonitoring.adjustTransactionTime(transaction)
+      expect(transaction._start).toBe(transactionStart)
+      expect(transaction._end).toBeGreaterThanOrEqual(longSpan._end)
+      done()
+    }
+    transaction.detectFinish()
+  })
+
+  it('should truncate active spans after transaction ends', () => {
+    const transaction = new Transaction('transaction', 'transaction')
+    const span = transaction.startSpan('test', 'test')
+    expect(transaction.spans.length).toBe(0)
+    expect(Object.keys(transaction._activeSpans).length).toBe(1)
+    transaction.end()
+
+    performanceMonitoring.adjustTransactionTime(transaction)
+    expect(transaction.spans.length).toBe(1)
+    expect(Object.keys(transaction._activeSpans).length).toBe(0)
+    expect(span.type).toContain('.truncated')
+  })
+
   it('should create correct payload', function() {
     var tr = new Transaction('transaction1', 'transaction1type', {
       transactionSampleRate: 1
     })
     var span = tr.startSpan('span1', 'span1type')
     span.end()
-    span._end += 10
```

Comment:
I think this was added to avoid having the span duration to be 0 since we start and end the span quickly.

Reply:
Yeah I had the same impression, It makes the span truncated since transaction end is lesser. But I can call performance.adjustTime and pass the test. We pay some time for creating the span so I thought its reasonable to remove it .WDYT?

Reply:
But `performanceMonitoring.createTransactionPayload` calls the adjustTime!

Also, it's ok to change the tr._end after `expect(tr._end).toBeDefined()` to make sure that it includes the span!

Reply:
Did the changes.